### PR TITLE
Bh/hang fix

### PR DESCRIPTION
**Pull request by bheni**
Fixes https://github.com/dolthub/dolt/issues/1153

and disables GC on import errors.

### CODE REVIEW

File: go/libraries/doltcore/mvdata/data_mover.go [hunk 1]
```diff
@@ -90,15 +90,26 @@ func (dmce *DataMoverCreationError) String() string {
...
+
 // Move is the method that executes the pipeline which will move data from the pipeline's source DataLocation to it's
 // dest DataLocation.  It returns the number of bad rows encountered during import, and an error.
 func (imp *DataMover) Move(ctx context.Context) (badRowCount int64, err error) {
 	defer imp.Rd.Close(ctx)
 	defer func() {
 		closeErr := imp.Wr.Close(ctx)
-		if closeErr != nil {
+		if err == nil {
 			err = closeErr
```
Comment (bheni) [inline go/libraries/doltcore/mvdata/data_mover.go:105]: I don't like overwriting an existing error if an error closing occurs.

Comment (Hydrocharged) [reply go/libraries/doltcore/mvdata/data_mover.go:105]: Yeah first error is probably the most important

**Review by Hydrocharged — APPROVED** [review]
Oh snap there's now video support for reviews! LGTM!

**Review by VinaiRachakonda — COMMENTED** [review]
If the changes are with cleaning up some channels, should we add a bats test.

**Comment by bheni** [comment]
@VinaiRachakonda Yeah, adding the case from the bug as a bats test would be good.

**Pull request merged**